### PR TITLE
Fix user search and add reset button

### PR DESCRIPTION
**Pull request by Lenninlasd**
## What does this PR do?

Fix user search and add reset button
### Screenshot

hpqnlovdy2
### Related Issue

https://github.com/terranodo/angular-search/issues/35

### CODE REVIEW

File: app/components/userFilter/userFilterDirective.js [hunk 1]
```diff
@@ -18,17 +18,24 @@
...
-                scope.userfilterInput = '';
+                scope.resetUser = resetUser;
 
+                scope.showUserFilterInfo = showUserFilterInfo;
                 /**
                  *
                  */
                 function userSearch() {
-                    HeatMapSourceGenerator.search(scope.userfilterInput);
+                    HeatMapSourceGenerator.filterObj.setUser(scope.userfilterInput);
```
Comment (milafrerichs) [inline app/components/userFilter/userFilterDirective.js]: please don't set the filterObj like this. use a method for it.
I realised this as well and added a `searchUser` method to `HeatMapSourceGenerator`

File: tests/service/HeatMapSourceGenerator.spec.js [hunk 2]
```diff
@@ -131,4 +131,28 @@ describe( 'HeatMapSourceGenerator', function() {
...
+            subject.filterObj.setUser('San Diego 2');
+            subject.filterObj.setTextDate('San Diego 3');
+            expect(subject.filterObj.getSearchObj()).toEqual(jasmine.objectContaining({
+                searchText: 'San Diego 1',
+                user: 'San Diego 2',
+                textDate: 'San Diego 3'
+            }));
+        });
+        it('All reset have to be null', function() {
+            subject.filterObj.setSearchText('');
```
Comment (milafrerichs) [inline tests/service/HeatMapSourceGenerator.spec.js]: please use convenience methods on `HeatMapSourceGenerator` for resetting the search. please don't use `filterObj` directly. I want to replace this eventually.

**Review by milafrerichs — CHANGES_REQUESTED** [review]

File: tests/service/HeatMapSourceGenerator.spec.js (hunk 2 above)
Comment (milafrerichs) [reply tests/service/HeatMapSourceGenerator.spec.js]: I'm working on something to replace the `filterObj` will be in the query serialization.

**Comment by Lenninlasd** [comment]
@milafrerichs, to avoid conflicts I'll merge with your branch, using your approach.

**Comment by ingenieroariel** [comment]
why do we need two reset buttons?  Shouldn't there just be one reset for the whole UI? 

/cc @gsolat

**Comment by Lenninlasd** [comment]
@ingenieroariel @gsolat  visually is frowned upon that a button in other component reset the user in the other side. If it is only one button, I think it must be in other way

**Comment by Lenninlasd** [comment]
@ingenieroariel @milafrerichs the reset the visual improvements are being done by @gsolat in other branch, what is needed to merge this PR?

**Comment by milafrerichs** [comment]
wait till the #44 PR is merged for updating the user search and reset. maybe even close this one and work from the other one.

**Comment by ingenieroariel** [comment]
as discussed, let's create a reset button at the level of the export one.

**Pull request closed without merging**